gnd/src/main/java/com/google/android/gnd/repository/local/Edit.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright 2019 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.android.gnd.repository.local;
+
+import static java8.util.J8Arrays.stream;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.room.ColumnInfo;
+import androidx.room.Ignore;
+import androidx.room.TypeConverter;
+import com.google.android.gnd.vo.Record.Value;
+
+/** Fields describing operation type, old, and new values, common to Feature and Record edits. */
+public class Edit {
+  public enum Type implements IntEnum {
+    UNKNOWN(0),
+    CREATE(1),
+    UPDATE(2),
+    DELETE(3);
+
+    private final int intValue;
+
+    Type(int intValue) {
+      this.intValue = intValue;
+    }
+
+    public int intValue() {
+      return intValue;
+    }
+
+    @TypeConverter
+    public static int toInt(@Nullable Edit.Type value) {
+      return IntEnum.fromIntEnum(value, UNKNOWN);
+    }
+
+    @NonNull
+    @TypeConverter
+    public static Edit.Type fromInt(int intValue) {
+      return IntEnum.toIntEnum(stream(values()), intValue, UNKNOWN);
+    }
+  }
+
+  @ColumnInfo(name = "type")
+  @NonNull
+  public Type type;
+
+  /** When null the edit refers to the entity itself. */
+  @ColumnInfo(name = "key")
+  @Nullable
+  public String key;
+
+  // TODO: Add Converter to convert to/from JSON.
+  @Ignore @Nullable public Value oldValue;
```

Comment:
Aren't the `Value` fields specific to record edits? Why put them on the shared `Edit` instead of `RecordEditEntity`?

Reply:
The idea was to use the same mechanism to represent edits on `Feature`s and `Record`s to a) allow code sharing and b) to facilitate adding custom attributes to Features and/or pre-defined attributes to Record in the future.

I've removed Value from the db persistence as per our other discussions and using a generic representation analogous to Firestore (JSON) for key-value pairs of changed values. Wdyt?

Reply:
Hmm, using the JSON means that each edit would store the entire state of the feature or record, right? Instead of just storing the values for the particular key that is to be updated. That could make merging logic trickier if we ever need to support concurrent modifications -- it isn't as clear to see which keys were explicitly modified in this edit, so we might need to overwrite the entire feature state (including keys that the user didn't explicitly set when creating the edit).

We might have discussed this issue before, but I don't remember what we decided :).